### ISSUE TITLE
Comm panel preview shows plain text as html (parameters only)

### ISSUE THREAD

**Issue by willgearty** [learning-unlimited/ESP-Website#2564]
If you don't put `<html>` tags around your email, it will not be sent in HTML format, but the email preview will still be in HTML format. Ideally, the commpanel preview should look exactly how the email will look in the user's inbox, which should include not formatting HTML code if there are no `<html>` tags.

Comm Panel Preview:
![image](https://user-images.githubusercontent.com/7232514/39138812-6e9e5380-46d5-11e8-95fc-0d501a1c863c.png)

In Gmail:
![image](https://user-images.githubusercontent.com/7232514/39138712-6547d4fa-46d5-11e8-97e4-5a98afa09b38.png)

**Comment by willgearty** [comment]
Ah, thanks for trying to reproduce. After trying to reproduce the behavior myself I realized that it's only if you use the various parameters (e.g. `{{program.teacher_schedule}}` in this case). If you just type random html it does not get formatted, but any code that is inserted by parameters ends up formatted in the preview, even without `<html>` tags.

**Comment by 900miles** [comment]
Yeah, that seems to be the case. It looks like the code for sanitizing html for the email preview is:
```
if '<html>' not in body:
    htmlbody = body.replace('<', '&lt;').replace('>', '&gt;').replace('\n', '<br />')
else:
    htmlbody = body
```
So I'm guessing the problem is that the code is imported by the parameters only after the code is sanitized, so the browser interprets it as real html code, but gmail doesn't.

I'm wondering if it would be more user-friendly to automatically correctly format parameter code as html code though, instead of trying to unformat it. That way people don't have to put \<html\> tags just to use parameters.

**Comment by willgearty** [comment]
Even worse, I think, is that that code would render html in the email preview for something like this:

`<html><h1>Hello</h1>`

But of course that's missing a `</html>` tag, which would be problematic once sent.

Aside from the existing edge-cases that that code doesn't address, are you suggesting that we add `<html>` tags around the email if parameters are included? Because that seems reasonable to me, but theoretically that could result in converting other plain text to html in the resulting email?

**Comment by 900miles** [comment]
Oh, right, it's pretty easy to forget adding a closing ``</html>`` tag, which could be a problem if the preview shows it as valid html. 

Adding ``<html>`` tags around the email should theoretically fix the problem, but as you said it might mess with the other text in the email. I'm not sure if there's a way to render only part of an email as html and leave the rest untouched.

Regarding #1898, if I'm understanding correctly, it will add two radio buttons for html or plaintext emails. The problem I see is if someone chooses plaintext, then uses a parameter. Then we're back to the same problem.

**Comment by willgearty** [comment]
Oh, I guess I'm not up to date on the details of #1898, so yeah that won't fix this. Would putting `<html>` tags just around the inserted parameter text work?

**Comment by 900miles** [comment]
I was thinking of doing something like that too, but the problem comes when there's two parameters, or if the user already put the ``<html>`` tags. It runs the risk of having multiple ``<html>`` tags in the same email, which apparently can cause problems. 

Maybe once #1898 is merged, we can make people use the html option if they want to use parameters. It looks like they already plan to show a warning if plaintext is selected and html elements (like ``<p>``) are detected, so maybe we can do the same if parameters are detected.

**Comment by willgearty** [comment]
Ah, yeah, then that won't work.
A check sounds like a good idea to me then. @milescalabresi could you incorporate that into #1898? Or would you prefer it be a separate thing?

**Comment by milescalabresi** [comment]
Yes, absolutely! That's a good thing to include. I will make sure it gets into the next commit. Trying to get that branch merged by the fall.

**Comment by willgearty** [comment]
I think #2666 will help solve this problem since all emails will now be html, essentially, and we can just add the `<html>` tags to every email (provided they aren't already included with the source code editor).

**Comment by willgearty** [comment]
Fixed by #2666.
